packages/discovery-provider/ddl/migrations/0102_shorten_challenge_ids.sql

```diff
@@ -0,0 +1,246 @@
+/** 
+ * Migrates our existing challenge IDs to shorter ones, and moves specifiers to 
+ * hex values so that they are shorter (when stringified) too.
+ * This allows us to put challenge claims into single Solana transactions as
+ * the disbursement IDs need to be less than 21 bytes.
+ * 
+ * This migration turns off triggers, drops indexes, and drops pkey constraints
+ * to increase speed, and carefully readds them in the end.
+**/
+
+BEGIN;
+
+/* CHALLENGES */
+
+-- remove foreign key
+ALTER TABLE user_challenges DROP CONSTRAINT user_challenges_challenge_id_fkey;
+
+-- Update challenges table
+UPDATE challenges SET id = 'p' WHERE id = 'profile-completion';
+UPDATE challenges SET id = 'l' WHERE id = 'listen-streak';
+UPDATE challenges SET id = 'u' WHERE id = 'track-upload';
+UPDATE challenges SET id = 'r' WHERE id = 'referrals';
+UPDATE challenges SET id = 'rv' WHERE id = 'ref-v';
```

Comment:
maybe we should just use one char for these. since we're going this far and they're already confusing.

Reply:
this is an annoying refactor ask tho, so please feel free to ignore